src/test_lib_json/main.cpp

```diff
@@ -1817,7 +1817,7 @@ JSONTEST_FIXTURE_LOCAL(ValueTest, StaticString) {
 
 JSONTEST_FIXTURE_LOCAL(ValueTest, WideString) {
   // https://github.com/open-source-parsers/jsoncpp/issues/756
-  const std::string uni = u8"\u5f0f\uff0c\u8fdb"; // "式，进"
+  const std::string uni = "式，进"; //u8"\u5f0f\uff0c\u8fdb"
```

Comment:
I don't understand the need to change these string literals. They were deliberately exposing the unicode in hex so we can check the output more readily.

Reply:
Okay. So i dug a bit deeper into this. And i think my fix is the wrong way of doing it.
What has happened is that u8 returned a std::string untill C++20 ([cppreference](https://en.cppreference.com/w/cpp/language/string_literal)). Now it returns a std::u8string ([cppreference](https://en.cppreference.com/w/cpp/string/basic_string))

Reply:
Oh okay. `u8` literals were redefined under C++20 to produce `char8_t[]` instead of `char[]`.
What we want is a `std::string` containing UTF-8 characters.

I believe we can cast the char8_t[] to const char* as size and alignment match.

```
const std::string uni = (const char*)u8"\u5f0f\uff0c\u8fdb"; // "式，进"
```

Reply:
Why not use auto?

Reply:
> Why not use auto?

Sorry, I don't know what you mean.
Why would we use `auto` for this? I mean, how would it help?

Reply:
avoid pointless conversions.